src/main/java/graphql/execution/directives/QueryDirectivesImpl.java

```diff
@@ -56,16 +62,32 @@ private void computeValuesLazily() {
 
             final Map<Field, List<GraphQLDirective>> byField = new LinkedHashMap<>();
             final Map<Field, List<QueryAppliedDirective>> byFieldApplied = new LinkedHashMap<>();
+
+            BiMap<GraphQLDirective, Directive> directiveCounterParts = HashBiMap.create();
+            BiMap<GraphQLDirective, QueryAppliedDirective> gqlDirectiveCounterParts = HashBiMap.create();
+            BiMap<QueryAppliedDirective, GraphQLDirective> gqlDirectiveCounterPartsInverse = gqlDirectiveCounterParts.inverse();
             mergedField.getFields().forEach(field -> {
                 List<Directive> directives = field.getDirectives();
                 ImmutableList<GraphQLDirective> resolvedDirectives = ImmutableList.copyOf(FpKit.flatList(
                         directivesResolver
-                                .resolveDirectives(directives, schema, variables, graphQLContext, locale)
+                                .resolveDirectives(directives, schema, coercedVariables, graphQLContext, locale)
                                 .values()
                 ));
+                for (int i = 0; i < directives.size(); i++) {
+                    Directive directive = directives.get(i);
+                    GraphQLDirective graphQLDirective = resolvedDirectives.get(i);
+                    directiveCounterParts.put(graphQLDirective, directive);
+                }
```

Comment:
Question - this is mapping the AST directive to its GraphQLDirective counter part by index - does directivesResolver return the list in the same order and size.